webpack/assets/javascripts/react_app/common/testHelpers.js

```diff
@@ -58,6 +60,30 @@ export const testComponentSnapshotsWithFixtures = (Component, fixtures) =>
       it(description, () => expect(toJson(component)).toMatchSnapshot())
   );
 
+const resolveDispatch = async (action, depth) => {
+  // if it is async action and we are allowed to go deeper
+  if (depth && typeof action === 'function') {
+    const dispatch = jest.fn();
+    await action(dispatch);
+    jest.runOnlyPendingTimers();
+
+    return Promise.all(
+      dispatch.mock.calls.map(call => resolveDispatch(call[0], depth - 1))
+    );
+  }
+  // else return the action itself
+  return action;
+};
+
+/**
+ * run an action (sync or async) and returns a call tree
+ * @param  {Function}  runAction  Action runner function
+ * @param  {Number} states the depth of dispatch calls
+ * @return calls result tree to the given depth - array for each branch of calls
+ */
+export const runActionInDepth = (runAction, depth = 1) =>
```

Comment:
@amirfefer what do you think about his one? I have figured, I would like to test deeper caller tree of the dispatch, not just the first level, so I wrote this helper for it.

Reply:
@sharvit ping